runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -679,6 +682,54 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		stringStack[stringStackSize - 1] += titleBuilder.toString();
 	}
 
+	private void setLimitReset(int itemId)
+	{
+		String lastLimit = getLimitReset(itemId);
+		if (!lastLimit.equals("None"))
+		{
+			return; // do not update if previous limit has not expired
+		}
+		String itemIdStr = Integer.toString(itemId);
+		Instant currentDateTime = Instant.now();
+		configManager.setConfiguration("grandexchange",
+			client.getUsername().toLowerCase() + itemIdStr, currentDateTime);
+	}
+
+	private String getLimitReset(int itemId)
+	{
+		String itemIdStr = Integer.toString(itemId);
+		Instant lastDateTime = configManager.getConfiguration("grandexchange",
+			client.getUsername().toLowerCase() + itemIdStr, Instant.class);
+		if (lastDateTime == null)
+		{
+			return "None";
+		}
+		Instant now = Instant.now();
+		long minuteDifference = ChronoUnit.MINUTES.between(lastDateTime, now);
+		if (minuteDifference / 60.0 > 4.0)
```

Comment:
Can't this just be `if (minuteDifference >= 240)`?

Reply:
yes, but i wanted it to be clearer. if someone were to read the code it would be more apparent it was referring to 4 hours

Reply:
Like this ?
```
		Duration difference = Duration.between(lastDateTime, now);
		Duration fourHours = Duration.ofHours(4);
		if (difference.compareTo(fourHours) > 0)
		{

		}
		else
		{
			long millisLeft = fourHours.minusMinutes(difference.toMinutes()).toMillis();
			System.out.println(DurationFormatUtils.formatDuration(millisLeft, "HH:mm"));
		}
```

Reply:
Thats pretty smart ngl. Is there any advantage of this method over building it manually, tho?

Reply:
Readability without comments and brain racking

Reply:
Yeah, learnt something today. Thanks! I'll use this implementation if @Adam- approves.